jax/_src/debugging.py

```diff
@@ -48,11 +53,21 @@ def debug_callback_abstract_eval(*flat_avals, callback: Callable[..., Any],
   del flat_avals, callback, in_tree
   return [], {effect}
 
-def debug_callback_batching_rule(*flat_args, callback: Callable[..., Any],
-    effect: DebugEffect, in_tree: tree_util.PyTreeDef):
-  del flat_args, callback, effect, in_tree
-  # TODO(sharadmv): implement batching rule
-  raise NotImplementedError('Batching not supported for `debug_callback`.')
+def debug_callback_batching_rule(args, dims, **params):
+  """Unrolls the debug callback across the mapped axis."""
+  axis_size = next(x.shape[i] for x, i in zip(args, dims)
+                   if i is not None)
+  def get_arg_at_dim(i, dim, arg):
+    if dim is None:
+      # Broadcast unmapped argument
+      return arg
+    return lax.index_in_dim(arg, i, axis=dim, keepdims=False)
+  outs = []
+  for i in range(axis_size):
```

Comment:
We could consider generating a rolled loop, i.e. a lax.map. That might avoid inducing large compile times. Can do that in a follow-up though.

Reply:
Added a TODO